docs/deployment/pull-backup.rst

```diff
@@ -174,3 +187,110 @@ directly extract it without the need of mounting with SSHFS:
 Note that in this scenario the tar format is the limiting factor – it cannot
 restore all the advanced features that BorgBackup supports. See
 :ref:`borg_export-tar` for limitations.
+
+socat
+=====
+
+The program socat has to be available on the backup server and on the machine
+to be backed up.
+
+Short Version
+-------------
+
+Execute the following command on the machine you want to store the backup::
+
+   socat UNIX-LISTEN:/tmp/borg_serve,fork EXEC:"borg serve --append-only --restrict-to-path /path_to_repository/"
```

Comment:
Also` /path/to` for the socket, or something sane on most systems like `/var/urn`. Plus the socket name should not collide with other such backups, so e.g. `/var/run/borg/reponame.sock`.

Reply:
I did only mention /path/to/repo as we use this consistently in the docs for Borg repo paths.

Reply:
When using `/var/run/borg/` for the socket on client and server side, we have to create the folder beforehand and give it sufficient rights to make sure the user on client and server side can create a file in this folder. Are the downsides of using `/tmp/` so heavy, to introduce those additional steps?

Reply:
Dealing with permissions is never about sufficient, but about adequate, as in no more than absolutely needed. A private path in` /var/run` that is **only** accessible by the borg backup user has adequate permissions. A **predictable** path in world writable` /tmp` may have sufficient permissions to "work", but it is not at all a case of adequate permissions, because it allows any other user on that system to e.g. carry out a symlink attack against that path. IOW you create a security issue for anyone following a documentation like that.
So either` /path/to` to force the user to think on his own and live with whatever comes out of that, or if using a real path, make it work correctly.

Unlike the idea for or hackish implementation of a workaround mentioned somewhere deeply buried in an issue tracker, a documentation like this targets people that do not know better. Either you require better knowledge by not documenting the detail of such advanced usage scenarios, or you document it in the most correct and bullet proof way to not harm such people that follow the outlined steps in good faith.

Reply:
Sounds fair.
I've added 6d837833 for this.
Or do you prefer `/path/to` and an additional paragraph to suggest `/var/run/`?

Reply:
No need for /path/to. Being the current thing in FHS just `/run` is fine I'd say. BSD users will have enough of a clue to know whether they still have `/var/urn` ro not. :wink:

Reply:
So I think we're done here? I resolve the conversation.